src/lib/propagation/node.js

```diff
@@ -0,0 +1,79 @@
+import * as constants from './constants'
+
+const {
+  NODE_TYPES,
+  nodeTypeToGuesstimateType,
+  ERROR_TYPES: {GRAPH_ERROR},
+  ERROR_SUBTYPES: {GRAPH_SUBTYPES: {MISSING_INPUT_ERROR, IN_INFINITE_LOOP, INVALID_ANCESTOR_ERROR}},
+} = constants
+
+import {Guesstimator} from 'lib/guesstimator/index'
+import {_matchingFormatter} from 'lib/guesstimator/formatter/index'
+
+import * as _collections from 'gEngine/collections'
+import * as _utils from 'gEngine/utils'
+
+export class SimulationNode {
+  constructor({id, expression, type, guesstimateType, samples, errors, childrenIndices, parentIndices, ancestors}, DAG, index) {
+    this.id = id
+    this.expression = expression
+    this.type = type
+    this.guesstimateType = guesstimateType
+    this.samples = samples
+    this.errors = errors
+    this.parentIndices = parentIndices
+    this.childrenIndices = childrenIndices
+    this.ancestors = ancestors
+    this.DAG = DAG
+    this.index = index
```

Comment:
It needs most of them. The childrenIndices aren't used presently, but may be needed later, if the graph gets too big. Should I take them out for now?